File: integration-tests/iac/lambdas/python/eventbridge_producer.py
import json
import os
import boto3

events = boto3.client('events')


def handler(event, context):
    event_bus_name = os.environ['EVENT_BUS_NAME']

    detail = {
        'message': 'Hello from EventBridge producer!',
        'requestId': context.aws_request_id,
    }

    response = events.put_events(
        Entries=[
            {
                'Source': 'tracing-tests.producer',
                'DetailType': 'TestMessage',
                'Detail': json.dumps(detail),
                'EventBusName': event_bus_name,
            }
        ]
    )

    print(f"Put event to EventBridge: {response}")

    return {
        'statusCode': 200,
        'body': json.dumps({
            'failed_entry_count': response['FailedEntryCount'],
        }),
    }

File: integration-tests/iac/lambdas/node/eventbridge_producer.mjs
import { EventBridgeClient, PutEventsCommand } from '@aws-sdk/client-eventbridge';

const events = new EventBridgeClient();

export async function handler(event, context) {
    const eventBusName = process.env.EVENT_BUS_NAME;

    const detail = {
        message: 'Hello from EventBridge producer!',
        requestId: context.awsRequestId,
    };

    const response = await events.send(new PutEventsCommand({
        Entries: [
            {
                Source: 'tracing-tests.producer',
                DetailType: 'TestMessage',
                Detail: JSON.stringify(detail),
                EventBusName: eventBusName,
            },
        ],
    }));

    console.log(`Put event to EventBridge: ${JSON.stringify(response)}`);

    return {
        statusCode: 200,
        body: JSON.stringify({
            failed_entry_count: response.FailedEntryCount,
        }),
    };
}
